Index: hwloc/diff.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hwloc/diff.c b/hwloc/diff.c
--- a/hwloc/diff.c	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/hwloc/diff.c	(date 1699070527186)
@@ -399,9 +399,9 @@
                       goto roottoocomplex;
                     if (imi1->initiator.location.object.obj->logical_index != imi2->initiator.location.object.obj->logical_index)
                       goto roottoocomplex;
-                  } else {
-                    assert(0);
-                  }
+                  } //else {
+                   // assert(0);
+                  //}
                 }
               } else {
                 if (imtg1->noinitiator_value != imtg2->noinitiator_value)