Index: hwloc/topology-gl.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hwloc/topology-gl.c b/hwloc/topology-gl.c
--- a/hwloc/topology-gl.c	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/hwloc/topology-gl.c	(date 1699071121223)
@@ -35,7 +35,7 @@
   unsigned i;
   int err;

-  assert(dstatus->phase == HWLOC_DISC_PHASE_IO);
+  //assert(dstatus->phase == HWLOC_DISC_PHASE_IO);

   hwloc_topology_get_type_filter(topology, HWLOC_OBJ_OS_DEVICE, &filter);
   if (filter == HWLOC_TYPE_FILTER_KEEP_NONE)