Subject: [PATCH] topology-aix.c.patch
---
Index: hwloc/topology-aix.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hwloc/topology-aix.c b/hwloc/topology-aix.c
--- a/hwloc/topology-aix.c	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/hwloc/topology-aix.c	(date 1699469968713)
@@ -535,7 +535,7 @@
   rsid_t rsid = { .at_subrange = &subrange };
   uint_t aix_policy;
   int ret;
-  fprintf(stderr,"yop\n");
+  // fprintf(stderr,"yop\n");
 
   if ((flags & (HWLOC_MEMBIND_MIGRATE|HWLOC_MEMBIND_STRICT))
             == (HWLOC_MEMBIND_MIGRATE|HWLOC_MEMBIND_STRICT)) {
@@ -626,8 +626,8 @@
     unsigned os_index = HWLOC_UNKNOWN_INDEX; /* no os_index except for PU and NUMANODE below */
 
     if (rs_getrad(rset, rad, sdl, i, 0)) {
-      if (HWLOC_SHOW_CRITICAL_ERRORS())
-        fprintf(stderr, "hwloc/aix: rs_getrad(%d) failed: %s\n", i, strerror(errno));
+      // if (HWLOC_SHOW_CRITICAL_ERRORS())
+      // fprintf(stderr, "hwloc/aix: rs_getrad(%d) failed: %s\n", i, strerror(errno));
       continue;
     }
     if (!rs_getinfo(rad, R_NUMPROCS, 0))