Subject: [PATCH] topology-rsmi.c.patch
---
Index: hwloc/topology-rsmi.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hwloc/topology-rsmi.c b/hwloc/topology-rsmi.c
--- a/hwloc/topology-rsmi.c	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/hwloc/topology-rsmi.c	(date 1699469791428)
@@ -91,11 +91,11 @@
   rsmi_status_t rsmi_rc = rsmi_dev_name_get(dv_ind, device_name, size);
 
   if (rsmi_rc != RSMI_STATUS_SUCCESS) {
-    if (HWLOC_SHOW_ALL_ERRORS()) {
-      const char *status_string;
-      rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
-      fprintf(stderr, "hwloc/rsmi: GPU(%u): Failed to get name: %s\n", (unsigned)dv_ind, status_string);
-    }
+    // if (HWLOC_SHOW_ALL_ERRORS()) {
+    //  const char *status_string;
+    //  rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
+      // fprintf(stderr, "hwloc/rsmi: GPU(%u): Failed to get name: %s\n", (unsigned)dv_ind, status_string);
+    //}
     return -1;
   }
   return 0;
@@ -112,11 +112,11 @@
   rsmi_status_t rsmi_rc = rsmi_dev_pci_id_get(dv_ind, bdfid);
 
   if (rsmi_rc != RSMI_STATUS_SUCCESS) {
-    if (HWLOC_SHOW_ALL_ERRORS()) {
-      const char *status_string;
-      rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
-      fprintf(stderr, "hwloc/rsmi: GPU(%u): Failed to get PCI Info: %s\n", (unsigned)dv_ind, status_string);
-    }
+    // if (HWLOC_SHOW_ALL_ERRORS()) {
+    //  const char *status_string;
+    //  rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
+    //  fprintf(stderr, "hwloc/rsmi: GPU(%u): Failed to get PCI Info: %s\n", (unsigned)dv_ind, status_string);
+    //}
     return -1;
   }
   return 0;
@@ -160,7 +160,7 @@
   if (rsmi_rc != RSMI_STATUS_SUCCESS) {
     return -1;
   }
-  sprintf(buffer, "%llx", (unsigned long long) id);
+  snprintf(buffer, sizeof(id), "%llx", (unsigned long long) id);
   return 0;
 }
 
@@ -193,14 +193,14 @@
   rsmi_status_t rsmi_rc = rsmi_dev_xgmi_hive_id_get(dv_ind, &hive_id);
 
   if (rsmi_rc != RSMI_STATUS_SUCCESS) {
-    if (HWLOC_SHOW_ALL_ERRORS()) {
-      const char *status_string;
-      rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
-      fprintf(stderr, "hwloc/rsmi: GPU(%u): Failed to get hive id: %s\n", (unsigned)dv_ind, status_string);
-    }
+    // if (HWLOC_SHOW_ALL_ERRORS()) {
+    //  const char *status_string;
+    //  rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
+    //  fprintf(stderr, "hwloc/rsmi: GPU(%u): Failed to get hive id: %s\n", (unsigned)dv_ind, status_string);
+    // }
     return -1;
   }
-  sprintf(buffer, "%llx", (unsigned long long) hive_id);
+  snprintf(buffer, sizeof(hive_id), "%llx", (unsigned long long) hive_id);
   return 0;
 }
 
@@ -218,11 +218,11 @@
                                                   hops, type);
 
   if (rsmi_rc != RSMI_STATUS_SUCCESS) {
-    if (HWLOC_SHOW_ALL_ERRORS()) {
-      const char *status_string;
-      rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
-      fprintf(stderr, "hwloc/rsmi: GPU(%u): Failed to get link type: %s\n", (unsigned)dv_ind_src, status_string);
-    }
+    // if (HWLOC_SHOW_ALL_ERRORS()) {
+    //  const char *status_string;
+    //  rsmi_rc = rsmi_status_string(rsmi_rc, &status_string);
+    //  fprintf(stderr, "hwloc/rsmi: GPU(%u): Failed to get link type: %s\n", (unsigned)dv_ind_src, status_string);
+    //}
     return -1;
   }
   return 0;
@@ -256,11 +256,11 @@
 
   ret = rsmi_init(0);
   if (RSMI_STATUS_SUCCESS != ret) {
-    if (HWLOC_SHOW_ALL_ERRORS()) {
-      const char *status_string;
-      rsmi_status_string(ret, &status_string);
-      fprintf(stderr, "hwloc/rsmi: Failed to initialize with rsmi_init(): %s\n", status_string);
-    }
+    // if (HWLOC_SHOW_ALL_ERRORS()) {
+    //  const char *status_string;
+    //  rsmi_status_string(ret, &status_string);
+    //  fprintf(stderr, "hwloc/rsmi: Failed to initialize with rsmi_init(): %s\n", status_string);
+    //}
     return 0;
   }
 
@@ -268,11 +268,11 @@
 
   ret = rsmi_num_monitor_devices(&nb);
   if (RSMI_STATUS_SUCCESS != ret || !nb) {
-    if (RSMI_STATUS_SUCCESS != ret && HWLOC_SHOW_ALL_ERRORS()) {
-      const char *status_string;
-      rsmi_status_string(ret, &status_string);
-      fprintf(stderr, "hwloc/rsmi: Failed to get number of devices with rsmi_num_monitor_devices(): %s\n", status_string);
-    }
+    // if (RSMI_STATUS_SUCCESS != ret && HWLOC_SHOW_ALL_ERRORS()) {
+    //  const char *status_string;
+    //  rsmi_status_string(ret, &status_string);
+    //  fprintf(stderr, "hwloc/rsmi: Failed to get number of devices with rsmi_num_monitor_devices(): %s\n", status_string);
+    //}
     goto out;
   }
 
@@ -347,7 +347,7 @@
           continue;
         if ((get_device_io_link_type(i, j, &type, &hops) == 0) &&
             (type == RSMI_IOLINK_TYPE_XGMI)) {
-          xgmi_peers_ptr += sprintf(xgmi_peers_ptr, "rsmi%u ", j);
+          xgmi_peers_ptr += snprintf(xgmi_peers_ptr, 8, "rsmi%u ", j);
           xgmi_bws[i*nb+j] = 100000; /* TODO: verify the XGMI version before putting 100GB/s here? */
           xgmi_hops[i*nb+j] = hops;
           got_xgmi_bws = 1;
