Index: hwloc/topology-noos.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hwloc/topology-noos.c b/hwloc/topology-noos.c
--- a/hwloc/topology-noos.c	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/hwloc/topology-noos.c	(date 1699070814861)
@@ -22,7 +22,7 @@
   struct hwloc_topology *topology = backend->topology;
   int64_t memsize;

-  assert(dstatus->phase == HWLOC_DISC_PHASE_CPU);
+  //assert(dstatus->phase == HWLOC_DISC_PHASE_CPU);

   if (!topology->levels[0][0]->cpuset) {
     int nbprocs;
