Index: hwloc/topology-freebsd.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hwloc/topology-freebsd.c b/hwloc/topology-freebsd.c
--- a/hwloc/topology-freebsd.c	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/hwloc/topology-freebsd.c	(date 1699411151523)
@@ -394,7 +394,7 @@
     return -1;
   }
   for(int i = 0 ; i < ndomains ; i++){
-    sprintf(nb, "%d:", i);
+      snprintf(nb, 32, "%d:", i);
     ptr = strstr(locality, nb);
     ptr = ptr + 2;
     for(int j = 0 ; j < ndomains ; j++){
