Subject: [PATCH] clean
---
Index: hwloc/topology-pci.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hwloc/topology-pci.c b/hwloc/topology-pci.c
--- a/hwloc/topology-pci.c	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/hwloc/topology-pci.c	(date 1699466777737)
@@ -20,7 +20,7 @@
 #include <stdio.h>
 #include <fcntl.h>
 #include <string.h>
-#include <assert.h>
+//#include <assert.h>
 #include <stdarg.h>
 #ifdef HWLOC_LINUX_SYS
 #include <dirent.h>
@@ -162,7 +162,7 @@
     hwloc_pci_get_names(topology);
     return 0;
   }
-  assert(dstatus->phase == HWLOC_DISC_PHASE_PCI);
+  //assert(dstatus->phase == HWLOC_DISC_PHASE_PCI);
 
   hwloc_debug("%s", "\nScanning PCI buses...\n");
 
@@ -175,9 +175,9 @@
   ret = pci_system_init();
   if (ret) {
     HWLOC_PCIACCESS_UNLOCK();
-    if (HWLOC_SHOW_ALL_ERRORS())
-      fprintf(stderr, "hwloc/pci: Failed to initialize libpciaccess with pci_system_init(): %d (%s)\n",
-              ret, strerror(errno));
+    // if (HWLOC_SHOW_ALL_ERRORS())
+      // fprintf(stderr, "hwloc/pci: Failed to initialize libpciaccess with pci_system_init(): %d (%s)\n",
+      //        ret, strerror(errno));
     return -1;
   }
 
@@ -205,8 +205,8 @@
 #ifndef HWLOC_HAVE_32BITS_PCI_DOMAIN
     if (domain > 0xffff) {
       static int warned = 0;
-      if (!warned && HWLOC_SHOW_ALL_ERRORS())
-	fprintf(stderr, "hwloc/pci: Ignoring PCI device with non-16bit domain.\nPass --enable-32bits-pci-domain to configure to support such devices\n(warning: it would break the library ABI, don't enable unless really needed).\n");
+    //  if (!warned && HWLOC_SHOW_ALL_ERRORS())
+	// fprintf(stderr, "hwloc/pci: Ignoring PCI device with non-16bit domain.\nPass --enable-32bits-pci-domain to configure to support such devices\n(warning: it would break the library ABI, don't enable unless really needed).\n");
       warned = 1;
       continue;
     }
