Index: hwloc/topology-darwin.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hwloc/topology-darwin.c b/hwloc/topology-darwin.c
--- a/hwloc/topology-darwin.c	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/hwloc/topology-darwin.c	(date 1699071121232)
@@ -571,7 +571,7 @@

   memset(&gothybrid, 0, sizeof(gothybrid));

-  assert(dstatus->phase == HWLOC_DISC_PHASE_CPU);
+  //assert(dstatus->phase == HWLOC_DISC_PHASE_CPU);

   if (topology->levels[0][0]->cpuset)
     /* somebody discovered things */