Subject: [PATCH] remove assert
---
Index: include/highfive/bits/H5Inspector_misc.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/include/highfive/bits/H5Inspector_misc.hpp b/include/highfive/bits/H5Inspector_misc.hpp
--- a/include/highfive/bits/H5Inspector_misc.hpp	(revision 88fcc898970f93a63be8e25760d6b9f33589690f)
+++ b/include/highfive/bits/H5Inspector_misc.hpp	(date 1699066612235)
@@ -376,7 +376,7 @@
         sizes[0] = val.size();
         if (!val.empty()) {
             auto s = inspector<value_type>::getDimensions(val[0]);
-            assert(s.size() + ndim == sizes.size());
+            //assert(s.size() + ndim == sizes.size());
             for (size_t i = 0; i < s.size(); ++i) {
                 sizes[i + ndim] = s[i];
             }
