Index: include/hwloc/bitmap.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/include/hwloc/bitmap.h b/include/hwloc/bitmap.h
--- a/include/hwloc/bitmap.h	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/include/hwloc/bitmap.h	(date 1699071311372)
@@ -15,7 +15,7 @@

 #include "hwloc/autogen/config.h"

-#include <assert.h>
+//#include <assert.h>


 #ifdef __cplusplus
@@ -386,7 +386,6 @@
  */
 #define hwloc_bitmap_foreach_begin(id, bitmap) \
 do { \
-        assert(hwloc_bitmap_weight(bitmap) != -1); \
         for (id = hwloc_bitmap_first(bitmap); \
              (unsigned) id != (unsigned) -1; \
              id = hwloc_bitmap_next(bitmap, id)) {