Index: include/hwloc/glibc-sched.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/include/hwloc/glibc-sched.h b/include/hwloc/glibc-sched.h
--- a/include/hwloc/glibc-sched.h	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/include/hwloc/glibc-sched.h	(date 1699071508962)
@@ -20,7 +20,7 @@
 #include "hwloc.h"
 #include "hwloc/helper.h"

-#include <assert.h>
+//#include <assert.h>

 #if !defined _GNU_SOURCE || (!defined _SCHED_H && !defined _SCHED_H_) || (!defined CPU_SETSIZE && !defined sched_priority)
 #error Please make sure to include sched.h before including glibc-sched.h, and define _GNU_SOURCE before any inclusion of sched.h
@@ -68,7 +68,7 @@
 #else /* !CPU_ZERO_S */
   unsigned cpu;
   CPU_ZERO(schedset);
-  assert(schedsetsize == sizeof(cpu_set_t));
+  //assert(schedsetsize == sizeof(cpu_set_t));
   hwloc_bitmap_foreach_begin(cpu, hwlocset)
     CPU_SET(cpu, schedset);
   hwloc_bitmap_foreach_end();
@@ -110,7 +110,7 @@
   /* sched.h does not support dynamic cpu_set_t (introduced in glibc 2.7),
    * assume we have a very old interface without CPU_COUNT (added in 2.6)
    */
-  assert(schedsetsize == sizeof(cpu_set_t));
+  //assert(schedsetsize == sizeof(cpu_set_t));
   for(cpu=0; cpu<CPU_SETSIZE; cpu++)
     if (CPU_ISSET(cpu, schedset))
       if (hwloc_bitmap_set(hwlocset, cpu) < 0)