Index: include/hwloc/plugins.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/include/hwloc/plugins.h b/include/hwloc/plugins.h
--- a/include/hwloc/plugins.h	(revision f262ce42ff5a007a8f247b546a6fc322b2922d1e)
+++ b/include/hwloc/plugins.h	(date 1699071508959)
@@ -533,7 +533,7 @@
 {
   enum hwloc_type_filter_e filter = HWLOC_TYPE_FILTER_KEEP_NONE;
   hwloc_topology_get_type_filter(topology, type, &filter);
-  assert(filter != HWLOC_TYPE_FILTER_KEEP_IMPORTANT); /* IMPORTANT only used for I/O */
+  //assert(filter != HWLOC_TYPE_FILTER_KEEP_IMPORTANT); /* IMPORTANT only used for I/O */
   return filter == HWLOC_TYPE_FILTER_KEEP_NONE ? 0 : 1;
 }
